tls/s2n_resume.c

```diff
@@ -924,7 +929,7 @@ int s2n_config_wipe_expired_ticket_crypto_keys(struct s2n_config *config, int8_t
     }
 
     uint64_t now;
-    POSIX_GUARD(config->wall_clock(config->sys_clock_ctx, &now));
+    POSIX_ENSURE(config->wall_clock(config->sys_clock_ctx, &now) >= S2N_SUCCESS, S2N_ERR_CANCELLED);
```

Comment:
We only call most callbacks in one place, but we call the wall_clock and monotonic_clock whenever we need time, which is apparently often.

Rather than relying on people calling the clock callbacks right every time, maybe we should wrap them in little helper methods. They could be called like "s2n_config_wall_clock" and "s2n_config_monotonic_clock".

If you don't want to make this improvement, that's fine-- you've fixed the immediate problem.